staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/metrics/metrics.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const (
+	namespace = "apiserver"
+	subsystem = "encryptionconfig_controller"
+)
+
+var (
+	encryptionConfigHotReloadFailureTotal = metrics.NewCounter(
+		&metrics.CounterOpts{
+			Namespace:      namespace,
+			Subsystem:      subsystem,
+			Name:           "encryption_config_hot_reload_failures_total",
+			Help:           "Total number of failed hot reloads of encryption configuration.",
+			StabilityLevel: metrics.ALPHA,
+		},
+	)
+
+	encryptionConfigHotReloadSuccessTotal = metrics.NewCounter(
+		&metrics.CounterOpts{
+			Namespace:      namespace,
+			Subsystem:      subsystem,
+			Name:           "encryption_config_hot_reload_success_total",
+			Help:           "Total number of successful hot reloads of encryption configuration.",
+			StabilityLevel: metrics.ALPHA,
+		},
+	)
+
+	encryptionConfigHotReloadLastSuccessTimestampSeconds = metrics.NewGauge(
+		&metrics.GaugeOpts{
+			Namespace:      namespace,
+			Subsystem:      subsystem,
+			Name:           "encryption_config_reload_last_success_timestamp_seconds",
+			Help:           "Timestamp of the last successful hot reload of encryption configuration.",
+			StabilityLevel: metrics.ALPHA,
+		},
```

Comment:
How about if this Gauge metric records last time in seconds when a reload was performed and its status {success, fail}?

Reply:
We want to record the timestamp for the last successful hot reload. If the reload fails, then we don't care about when it failed. We just increment the failure count.  Is there any benefit to record a timestamp for failure as well?

Reply:
If this metric is used to create alerts, then having the timestamp for the last failure is also helpful to know how fresh this failure is.

Reply:
makes sense. updated.